crates/commitlog/src/buf.rs

```diff
@@ -0,0 +1,114 @@
+use std::{
+    alloc,
+    num::NonZeroUsize,
+    ops::{Deref, DerefMut},
+    ptr,
+};
+
+/// A byte buffer of non-zero size and proper alignment.
+#[derive(Debug)]
+pub struct Aligned {
+    data: ptr::NonNull<u8>,
+    layout: alloc::Layout,
+    size: usize,
+}
```

Comment:
Why is this parametric over the layout at runtime, rather than using const generics? Do we ever need to dynamically compute the layout?

Reply:
Hadn’t thought of const generics. I need to check for zeroes then, no?